geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/locator/GMSLocator.java

```diff
@@ -449,15 +450,21 @@ boolean recoverFromFile(File file) {
 
       int version = ois.readInt();
       int currentVersion = Version.getCurrentVersion().ordinal();
-      DataInputStream input = new DataInputStream(fileInputStream);
-      if (version != currentVersion) {
+      DataInputStream input;
+      if (version == currentVersion) {
+        input = new DataInputStream(ois);
+      } else if (version > currentVersion) {
+        return false;
+      } else {
         Version geodeVersion = Version.fromOrdinalNoThrow((short) version, false);
         logger.info("Peer locator found that persistent view was written with version {}",
             geodeVersion);
-        if (version > currentVersion) {
-          return false;
+        if (geodeVersion.equals(Version.GEODE_1_11_0)) {
```

Comment:
Maybe flip this equals check `Version.GEODE_1_11_0.equals(geodeVersion)` - I think fromOrdinalNoThrow could return null if the version is not found?